kroxylicious/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -84,10 +85,12 @@ public class KafkaProxyFrontendHandler
     private String sniHostname;
 
     private ChannelHandlerContext inboundCtx;
-    // The message buffered while we connect to the outbound cluster
-    // There can only be one such because auto read is disabled until outbound
+
+    // Messages buffered while we connect to the outbound cluster
+    // The size should be limited because auto read is disabled until outbound
     // channel activation
-    private Object bufferedMsg;
+    private List<Object> bufferedMsgs = new ArrayList<>();
```

Comment:
Would it be worth wrapping this in a class? 

I was wondering about the benefits of making this lazily allocated and then conditional handling that might be needed later. Which lead me to thinking about separation of concerns and to thinking that buffering might not really be a concern for the `KafkaProxyFrontendHandler`. Or maybe more accurately that its not just a concern of the KafkaProxyFrontendHandler